pyOCD/tools/pyocd.py

```diff
@@ -69,207 +76,490 @@ def dumpHexData(data, startAddress=0, width=8):
                     break
         print
 
-
 class ToolError(Exception):
     pass
 
+def cmdoptions(opts):
+    def process_opts(fn):
+        parser = optparse.OptionParser(add_help_option=False)
+        for opt in opts:
+            parser.add_option(opt)
+        def foo(inst, args):
+            namespace, other_args = parser.parse_args(args)
+            return fn(inst, namespace, other_args)
+        return foo
+    return process_opts
+
+class PyOCDConsole(object):
+    PROMPT = '>>> '
+
+    def __init__(self, tool):
+        self.tool = tool
+        self.last_command = ''
+
+    def run(self):
+        try:
+            while True:
+                try:
+                    line = raw_input(self.PROMPT)
+                    line = line.strip()
+                    if line:
+                        self.process_command_line(line)
+                        self.last_command = line
+                    elif self.last_command:
+                        self.process_command(self.last_command)
+                except KeyboardInterrupt:
+                    print
```

Comment:
When you type Ctrl-D to quit on a Posix system, the shell's prompt would appear on the same line as the last pyocd-tool prompt. This print prevents that. I added a comment and check so it doesn't print on Windows.